File: Data Structures & Algorithms/string-encode-and-decode/submission-9.py
class Solution:
    hm = {}
    # use the join function
    def encode(self, strs: List[str]) -> str:
        words = []
        for i in range(len(strs)):
            words.append(strs[i])
            self.hm[i] = strs[i]
        encoded_str = "".join(words)
        return encoded_str
        

    def decode(self, s: str) -> List[str]:
        res = []
        for i in range(len(self.hm)):
            res.append(self.hm[i]);
        self.hm.clear()
        return res
        

File: Data Structures & Algorithms/anagram-groups/submission-9.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        const wordMap = new Map();
        const res = new Array();
        for(let str of strs) {
            let temp = Array.from(str);
            temp.sort();
            let sortedStr = temp.join("");
            wordMap.set(sortedStr, []);
           
        }

        for(let str of strs) {
            let temp = Array.from(str);
            temp.sort();
            let sortedStr = temp.join("");
            const arr = wordMap.get(sortedStr);
            arr.push(str);
            wordMap.set(sortedStr, arr);
        }

        for(const val of wordMap.values()) {
            res.push(val);
        }

        return res;

    }
}
